### PR TITLE
Fixed linting warnings

### PR DESCRIPTION
**Pull request by jwair**
Fixing lint warnings, simplifying parts of the code.

### CODE REVIEW

File: config/auth.go [hunk 1]
```diff
@@ -65,9 +65,7 @@ func (c *AuthConfig) Merge(o *AuthConfig) *AuthConfig {
 // Finalize ensures there no nil pointers.
 func (c *AuthConfig) Finalize() {
 	if c.Enabled == nil {
-		c.Enabled = Bool(false ||
```
Comment (jwair) [inline config/auth.go:68]: Redundant logical OR with false
(This comment applies to multiple changes in this PR)

File: config/buffer_period.go [hunk 2]
```diff
@@ -6,8 +6,8 @@ import (
 )
 
 var (
-	DefaultBufferPeriodMin = time.Duration(5 * time.Second)
```
Comment (jwair) [inline config/buffer_period.go:9]: Redundant type casting
(This comment applies to multiple changes in this PR)

File: config/condition_no.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ func (c *NoConditionConfig) Copy() MonitorConfig {
 // configuration, with values in the other configuration taking precedence.
 // Maps and slices are merged, most other values are overwritten. Complex
 // structs define their own merge functionality.
-func (c *NoConditionConfig) Merge(o MonitorConfig) MonitorConfig {
+func (c *NoConditionConfig) Merge(_ MonitorConfig) MonitorConfig {
```
Comment (jwair) [inline config/condition_no.go:23]: explicitly discard passed in parameter

File: config/provider.go [hunk 4]
```diff
@@ -35,11 +35,11 @@ func (c *TerraformProviderConfigs) Copy() *TerraformProviderConfigs {
 
 	o := make(TerraformProviderConfigs, c.Len())
 	for i, t := range *c {
-		copy := make(TerraformProviderConfig)
```
Comment (jwair) [inline config/provider.go:38]: `copy` is a reserved word (builtin function)

File: controller/condition_monitor.go [hunk 5]
```diff
@@ -55,7 +55,7 @@ func (cm *ConditionMonitor) WatchDep(ctx context.Context) error {
 			// stop for context canceled
 			return ctx.Err()
 		}
-		cm.logDepSize(50, int64(ix))
```
Comment (jwair) [inline controller/condition_monitor.go:58]: `ix` is already an int64

File: controller/daemon.go [hunk 6]
```diff
@@ -114,7 +114,7 @@ func (ctrl *Daemon) Run(ctx context.Context) error {
 			&registration.ServiceRegistrationManagerConfig{
 				ID:                  *conf.ID,
 				Port:                *conf.Port,
-				TLSEnabled:          (conf.TLS != nil && *conf.TLS.Enabled),
```
Comment (jwair) [inline controller/daemon.go:117]: Redundant parenthesis

File: controller/driver_factory.go [hunk 7]
```diff
@@ -152,7 +152,7 @@ func newDriverFunc(conf *config.Config) (driverFactoryFunc, error) {
 
 // newTerraformDriver maps user configuration to initialize a Terraform driver
 // for a task
-func newTerraformDriver(ctx context.Context, conf *config.Config, task *driver.Task, w templates.Watcher) (driver.Driver, error) {
```
Comment (jwair) [inline controller/driver_factory.go:155]: Explicitly discard passed-in context, not used in the function

File: controller/inspect_test.go [hunk 8]
```diff
@@ -265,11 +265,11 @@ func testInspect(t *testing.T, numTasks int, setupNewDriver func(*driver.Task) d
 }
 
 // inspectMockDriver mocks the driver with the methods needed for inspect-mode
-func inspectMockDriver(task *driver.Task, inspecTaskErr error) driver.Driver {
```
Comment (jwair) [inline controller/inspect_test.go:268]: `task` is not used in the function

File: controller/inspect_test.go [hunk 9]
```diff
@@ -33,15 +33,15 @@ func Test_Inspect_Run(t *testing.T) {
 			"one task",
 			1,
 			func(task *driver.Task) driver.Driver {
-				return inspectMockDriver(task, nil)
```
Comment (jwair) [inline controller/inspect_test.go:36]: Function no longer takes `task` as an argument

File: controller/inspect_test.go [hunk 10]
```diff
@@ -265,11 +265,11 @@ func testInspect(t *testing.T, numTasks int, setupNewDriver func(*driver.Task) d
 }
 
 // inspectMockDriver mocks the driver with the methods needed for inspect-mode
-func inspectMockDriver(task *driver.Task, inspecTaskErr error) driver.Driver {
+func inspectMockDriver(inspectTaskErr error) driver.Driver {
 	d := new(mocksD.Driver)
 	d.On("RenderTemplate", mock.Anything).Return(true, nil)
 	d.On("InitTask", mock.Anything, mock.Anything).Return(nil).Once()
-	d.On("InspectTask", mock.Anything).Return(driver.InspectPlan{}, inspecTaskErr)
+	d.On("InspectTask", mock.Anything).Return(driver.InspectPlan{}, inspectTaskErr)
```
Comment (jwair) [inline controller/inspect_test.go:272]: Typo in variable name

File: controller/tasks_manager.go [hunk 11]
```diff
@@ -428,11 +428,11 @@ func (tm *TasksManager) TaskRunNow(ctx context.Context, taskName string) error {
 // TaskByTemplate returns the name of the task associated with a template id.
 // If no task is associated with the template id, returns false.
 func (tm TasksManager) TaskByTemplate(tmplID string) (string, bool) {
-	driver, ok := tm.drivers.GetTaskByTemplate(tmplID)
```
Comment (jwair) [inline controller/tasks_manager.go:431]: `driver` conflicts with package name

File: testutils/sdk/sdk.go [hunk 12]
```diff
@@ -36,10 +36,10 @@ func AddCheck(s *testutil.TestServer, t testing.TB, name, serviceID, status stri
 	}
 	put(s, t, "/v1/agent/check/register", payload)
 
-	UpdateCheck(s, t, name, serviceID, status)
+	UpdateCheck(s, t, name, status)
 }
 
-func UpdateCheck(s *testutil.TestServer, t testing.TB, name, serviceID, status string) {
```
Comment (jwair) [inline testutils/sdk/sdk.go:42]: `serviceID` is not used

File: testutils/consul.go [hunk 13]
```diff
@@ -320,7 +320,7 @@ func generateServices(n int, namefmt, idfmt fmtFunc) []testutil.TestService {
 			Name:    namefmt(i),
 			ID:      idfmt(i),
 			Address: "127.0.0.2",
-			Port:    int(baseport + i),
```
Comment (jwair) [inline testutils/consul.go:323]: Already an int

File: templates/tftmpl/template_catalog_services.go [hunk 14]
```diff
@@ -36,7 +36,7 @@ func (t CatalogServicesTemplate) RendersVar() bool {
 	return t.RenderVar
 }
 
-func (m CatalogServicesTemplate) appendModuleAttribute(body *hclwrite.Body) {
```
Comment (jwair) [inline templates/tftmpl/template_catalog_services.go:39]: Unified receiver name with other methods

File: driver/terraform.go [hunk 15]
```diff
@@ -443,12 +443,12 @@ func (tf *Terraform) initTask(ctx context.Context) error {
 }
 
 // deregisterTemplate attempts to deregister the hashicat template
-func (tf *Terraform) deregisterTemplate(ctx context.Context) {
```
Comment (jwair) [inline driver/terraform.go:446]: `ctx` is not needed

File: driver/terraform.go [hunk 16]
```diff
@@ -238,8 +238,8 @@ func (tf *Terraform) RenderTemplate(ctx context.Context) (bool, error) {
 	}
 
 	tf.logger.Trace("checking dependency changes for task", taskNameLogKey, taskName)
-	re, err := tf.renderTemplate(ctx)
```
Comment (jwair) [inline driver/terraform.go:241]: `renderTemplate()` no longer takes a context argument

File: driver/terraform.go [hunk 17]
```diff
@@ -168,11 +168,11 @@ func (tf *Terraform) InitTask(ctx context.Context) error {
 }
 
 // DestroyTask destroys task dependencies so that it is safe for deletion
-func (tf *Terraform) DestroyTask(ctx context.Context) {
+func (tf *Terraform) DestroyTask(_ context.Context) {
 	tf.mu.Lock()
 	defer tf.mu.Unlock()
 
-	tf.deregisterTemplate(ctx)
+	tf.deregisterTemplate()
```
Comment (jwair) [inline driver/terraform.go:175]: `renderTemplate()` no longer takes a context argument

File: driver/task_test.go [hunk 18]
```diff
@@ -244,17 +244,17 @@ func TestTask_ProviderIDs(t *testing.T) {
 
 	cases := []struct {
 		name     string
-		task     Task
+		task     *Task
```
Comment (jwair) [inline driver/task_test.go:247]: `Task` contains a mutex. When we loop through cases later in the test, the loop creates copies of the cases, including copying the `Task` object and its mutex. We should not copy locks (e.g. mutex) by value.

File: driver/terraform_test.go [hunk 19]
```diff
@@ -1103,7 +1103,7 @@ func TestGetServicesMetaData(t *testing.T) {
 			},
 			func() *tmplfunc.ServicesMeta {
 				sm := &tmplfunc.ServicesMeta{}
-				sm.SetMeta(meta)
+				_ = sm.SetMeta(meta)
```
Comment (jwair) [inline driver/terraform_test.go:1106]: Explicitly discard returned error from `SetMeta()`

**Review by mkam — APPROVED** [review]
Looks good to me, thanks for all the explanation comments!

**Pull request merged**